config/samples/operator_v1alpha1_externaldns.yaml

```diff
@@ -1,5 +1,15 @@
 apiVersion: externaldns.olm.openshift.io/v1alpha1
 kind: ExternalDNS
 metadata:
-  name: externaldns-sample
+  name: sample
+  namespace: openshift-external-dns-operator
 spec:
+  provider:
+    type: AWS
+    aws:
+      credentials:
+        name: aws-access-key
+        namespace: openshift-external-dns-operator
+  source:
+    type: Service
+  zones: ["dnszone"]
```

Comment:
Would it be better to leave `spec.zones` unspecified?  On platforms where zones can be autodetected, leaving it unspecified will allow this sample to work out of the box, if the credentials can be configured appropriately.  On other platforms, the sample won't work anyway.

Reply:
What I had in my mind is more like having the sample which encourages the user to think about where the records should be published. The reason for this is when I first managed to get all the credentials setup ExternalDNS started to try to publish DNS records for all the services in the whole cluster and I was rather thankful of having the dns zone which didn't match any real zone ID. Once the user feels more comfortable with how ExternalDNS works, the filters can be removed.
I actually added `namespace` and `annotationFilter` to the sample too, for the same reason.

Reply:
That makes sense.  The README suggested running `oc apply -k config/samples`, so I figured the sample was supposed to be ready to use out of the box, but it makes sense not to have the out-of-the-box example automatically do _too_ much.